dcc-lsystem/src/renderer.rs

```diff
@@ -465,3 +489,34 @@ impl<Q: TurtleContainer> Renderer<ImageRendererOptions> for TurtleRenderer<Q> {
         buffer
     }
 }
+
+#[cfg(feature = "data_renderer")]
+#[derive(Default)]
+/// A version of ImageRendererOptionsBuilder but intended for data only rendering (no image).
+/// For symmetry reasons and future proofing, it is implemented as an empty struct.
+pub struct DataRendererOptionsBuilder {}
+
+#[cfg(feature = "data_renderer")]
+impl DataRendererOptionsBuilder {
+    pub fn build(&mut self) -> DataRendererOptions {
+        DataRendererOptions {}
+    }
+}
+
+#[cfg(feature = "data_renderer")]
+/// A version of ImageRendererOptions but intended for data only rendering (no image).
+/// For symmetry reasons and future proofing, it is implemented as an empty struct.
+pub struct DataRendererOptions {}
+
+#[cfg(feature = "data_renderer")]
+impl<Q: TurtleContainer> Renderer<DataRendererOptions> for TurtleRenderer<Q> {
+    type Output = Vec<(i32, i32, i32, i32)>;
+
+    fn render(mut self, system: &LSystem, _options: &DataRendererOptions) -> Self::Output {
+        // Setup our state machine based on the LSystem state
+        self.compute(system.get_state());
+
+        // TODO: find a way to move lines() instead of cloning it with to_vec()
+        self.state.inner().inner().lines().to_vec()
```

Comment:
Hmm this is perhaps an unfortunate consequence of the way I set up the traits, doesn't look like theres an easy way to do this

Reply:
So.. keep the comment as a future reminder?

Reply:
I had an idea that the DataRenderer could return a ref iterator, but I could manage to get a lifetime spec into the `Output` type.

Reply:
Yeah keeping it is fine.